src/contracts/BCoWPool.sol

```diff
@@ -0,0 +1,227 @@
+   * verification function.
+   */
+  function commit(bytes32 orderHash) external {
+    if (msg.sender != address(SOLUTION_SETTLER)) {
+      revert CommitOutsideOfSettlement();
+    }
+    assembly ("memory-safe") {
+      tstore(COMMITMENT_SLOT, orderHash)
+    }
+  }
+
+  /**
+   * @inheritdoc IERC1271
+   */
+  function isValidSignature(bytes32 _hash, bytes memory signature) external view returns (bytes4) {
+    (GPv2Order.Data memory order) = abi.decode(signature, (GPv2Order.Data));
+
+    // TODO: unify error log criteria with BalV1 (use require and string error message)
+    if (appDataHash != hash(order.appData)) revert AppDataDoNotMatchHash();
+    bytes32 orderHash = order.hash(SOLUTION_SETTLER_DOMAIN_SEPARATOR);
+    if (orderHash != _hash) revert OrderDoesNotMatchMessageHash();
+
+    _requireMatchingCommitment(orderHash);
+
+    verify(order);
+
+    // A signature is valid according to EIP-1271 if this function returns
+    // its selector as the so-called "magic value".
+    return this.isValidSignature.selector;
+  }
+
+  function commitment() public view returns (bytes32 value) {
+    assembly ("memory-safe") {
+      value := tload(COMMITMENT_SLOT)
+    }
+  }
+
+  /**
+   * @notice This function checks that the input order is admissible for the
+   * constant-product curve for the given trading parameters.
+   * @param order `GPv2Order.Data` of a discrete order to be verified.
+   */
+  function verify(GPv2Order.Data memory order) public view {
+    Record memory inRecord = _records[address(order.sellToken)];
+    Record memory outRecord = _records[address(order.buyToken)];
+
+    require(inRecord.bound, 'BCoWPool: TOKEN_NOT_BOUND');
+    require(outRecord.bound, 'BCoWPool: TOKEN_NOT_BOUND');
+
+    require(order.validTo < block.timestamp + MAX_ORDER_DURATION, 'BCoWPool: ORDER_EXPIRED');
+    require(order.feeAmount == 0, 'BCoWPool: FEE_NOT_ZERO');
+    require(order.kind == GPv2Order.KIND_SELL, 'BCoWPool: INVALID_OPERATION');
+    require(order.buyTokenBalance == GPv2Order.BALANCE_ERC20, 'BCoWPool: INVALID_BALANCE');
+    require(order.sellTokenBalance == GPv2Order.BALANCE_ERC20, 'BCoWPool: INVALID_BALANCE');
+
+    uint256 tokenAmountOut = calcOutGivenIn({
+      tokenBalanceIn: order.sellToken.balanceOf(address(this)),
+      tokenWeightIn: inRecord.denorm,
+      tokenBalanceOut: order.buyToken.balanceOf(address(this)),
+      tokenWeightOut: outRecord.denorm,
+      tokenAmountIn: order.sellAmount,
+      swapFee: 0
+    });
+
+    require(tokenAmountOut >= order.buyAmount, 'BCoWPool: INSUFFICIENT_OUTPUT_AMOUNT');
+  }
+
+  /**
+   * @dev Computes an identifier that uniquely represents the parameters in
+   * the function input parameters.
+   * @param appData Bytestring that decodes to `AppData`
+   * @return The hash of the input parameter, intended to be used as a unique identifier
+   */
+  function hash(bytes32 appData) public pure returns (bytes32) {
+    return keccak256(abi.encode(appData));
+  }
```

Comment:
we use this only once, let's replace it for `keccak256(abi.encode(order.appData))`
less methods to write and test